time_series_transformer/transformer.py

```diff
@@ -51,39 +54,39 @@ class Transformer(nn.Module):
     """
 
     def __init__(self,
-                 d_input: int,
-                 d_model: int,
-                 d_output: int,
-                 q: int,
-                 v: int,
-                 h: int,
-                 N: int,
-                 attention_size: int = None,
-                 dropout: float = 0.3,
-                 chunk_mode: bool = True,
+                 input_dim: int = 1,
+                 hidden_dim: int = 32,
```

Comment:
I would like to keep ``d_model`` as an input parameter, for compatibility with the [pytorch Transformer](https://pytorch.org/docs/stable/generated/torch.nn.Transformer.html).

Reply:
The reason why I changed this was because in [TimeSeriesPredictor](https://time-series-predictor.readthedocs.io/en/latest/), I developed some handy technologies to ease development for data science developers. One of these technologies is dynamically setting the input size. More info about how it was implemented can be found [here](https://skorch.readthedocs.io/en/stable/user/FAQ.html?highlight=module__input_dim#how-can-i-dynamically-set-the-input-size-of-the-pytorch-module-based-on-the-data).

Among other features already present in TimeSeriesPredictor are:
- L1 regularization
- L2 regularization needs to be manually made inside the model (Transformer, or LSTM).
- Skorch
- Auto scaling of input and output
- Early stopping
- R2_score calculation

More recently I am evaluating to [add ray[tune] and tune_sklearn](https://github.com/ray-project/tune-sklearn/issues/153) to automatically optimize the network hyperparameters.

Reply:
In the case you still prefer to stay with the same names, I would actually have to create an specific TimeSeriesPredictor for the Transformer with appropriate names to allow dynamically setting input sizes correctly.

Reply:
I secured time_series_transformer package name, I can hand that over to you. Just send me an email at daniel(at)kryptonunite(dot)com. We can better discuss development and academic cooperation by email if you will.

Reply:
In that case, we may add an option at some point to dynamically set the input size. In the mean time, that doesn't mean we can't match pytorch's writing conventions !